### ISSUE TITLE
Test request throws exception with cloudformation-cli 0.1.6.  Fails to parse new fields.

### ISSUE THREAD

**Issue by drch-** [aws-cloudformation/cloudformation-cli-python-plugin#109]
https://github.com/aws-cloudformation/cloudformation-cli/commit/b8466befe9dffa82f5d6bcb3c071709ceff77163 added new fields to the test event, which are causing an exception when running `cfn test`

```
START RequestId: af819c5a-99f5-1d9c-ad69-7614c2c1d235 Version: $LATEST
[ERROR]	2020-08-07T21:56:25.652Z	af819c5a-99f5-1d9c-ad69-7614c2c1d235	Invalid request
Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 103, in _parse_test_request
    **event.request
TypeError: __init__() got an unexpected keyword argument 'region'
[ERROR]	2020-08-07T21:56:25.653Z	af819c5a-99f5-1d9c-ad69-7614c2c1d235	Handler error
Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 103, in _parse_test_request
    **event.request
TypeError: __init__() got an unexpected keyword argument 'region'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 119, in test_entrypoint
    session, request, action, callback_context = self._parse_test_request(event)
  File "/var/task/cloudformation_cli_python_lib/resource.py", line 110, in _parse_test_request
    raise InternalFailure(f"{e} ({type(e).__name__})") from e
cloudformation_cli_python_lib.exceptions.InternalFailure: __init__() got an unexpected keyword argument 'region' (TypeError)
END RequestId: af819c5a-99f5-1d9c-ad69-7614c2c1d235
```

**Comment by wnkz** [comment]
It looks like the wrong `cloudformation-cli-python-lib` is packaged in the build directory after running `cfn submit --dry-run`.

For example, after build if I check in `build/cloudformation_cli_python_lib/utils.py` I can see that the class `UnmodelledRequest` does not match current version.


```python
@dataclass
class UnmodelledRequest:
    clientRequestToken: str
    desiredResourceState: Optional[Mapping[str, Any]] = None
    previousResourceState: Optional[Mapping[str, Any]] = None
    logicalResourceIdentifier: Optional[str] = None
    nextToken: Optional[str] = None

    def to_modelled(self, model_cls: Type[BaseModel]) -> BaseResourceHandlerRequest:
        # pylint: disable=protected-access
        return BaseResourceHandlerRequest(
            clientRequestToken=self.clientRequestToken,
            desiredResourceState=model_cls._deserialize(self.desiredResourceState),
            previousResourceState=model_cls._deserialize(self.previousResourceState),
            logicalResourceIdentifier=self.logicalResourceIdentifier,
            nextToken=self.nextToken,
        )
```

This might be caused by the way the libraries are used / deployed. We have cloudformation-cli, cloudformation-cli-python-plugin and cloudformation-cli-python-lib and the latter does not seem to have a new version on pypi.

Did the maintainers forgot to make a new release for this package ? Version in setup.py is still 2.0.0 by the way https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/d00dd26c9f7a4a1baf3f134a3a12f4daac19d775/src/setup.py#L6

ping @PatMyron ? 🙏

**Comment by jotompki** [comment]
> It looks like the wrong `cloudformation-cli-python-lib` is packaged in the build directory after running `cfn submit --dry-run`.
> 
> For example, after build if I check in `build/cloudformation_cli_python_lib/utils.py` I can see that the class `UnmodelledRequest` does not match current version.
> 
> ```python
> @dataclass
> class UnmodelledRequest:
>     clientRequestToken: str
>     desiredResourceState: Optional[Mapping[str, Any]] = None
>     previousResourceState: Optional[Mapping[str, Any]] = None
>     logicalResourceIdentifier: Optional[str] = None
>     nextToken: Optional[str] = None
> 
>     def to_modelled(self, model_cls: Type[BaseModel]) -> BaseResourceHandlerRequest:
>         # pylint: disable=protected-access
>         return BaseResourceHandlerRequest(
>             clientRequestToken=self.clientRequestToken,
>             desiredResourceState=model_cls._deserialize(self.desiredResourceState),
>             previousResourceState=model_cls._deserialize(self.previousResourceState),
>             logicalResourceIdentifier=self.logicalResourceIdentifier,
>             nextToken=self.nextToken,
>         )
> ```
> 
> This might be caused by the way the libraries are used / deployed. We have cloudformation-cli, cloudformation-cli-python-plugin and cloudformation-cli-python-lib and the latter does not seem to have a new version on pypi.
> 
> Did the maintainers forgot to make a new release for this package ? Version in setup.py is still 2.0.0 by the way
> 
> https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/d00dd26c9f7a4a1baf3f134a3a12f4daac19d775/src/setup.py#L6
> 
> ping @PatMyron ? 🙏

It looks as if we haven't added these fields into the test request that is used when invoking the contract test specific entrypoint:

https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/797b9b3ae73fcb0002ee69b4a348842e007ddcab/src/cloudformation_cli_python_lib/resource.py#L91-L100

TestEvent here is missing the options passed in by contract tests:
https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/64f454e76021c60f9db4dfad815c6bd505cbbb48/src/cloudformation_cli_python_lib/utils.py#L31-L37

We need to add region, partition, and account to this request to this event so that the test requests can properly deserialize and release a new version. As a mitigation to unblock testing, you can install cloudformation-cli version 0.1.5 or below where these keys were not included in the test request as they were added in 0.1.6: https://github.com/aws-cloudformation/cloudformation-cli/releases/tag/v0.1.6. We will look into getting these added so the latest cloudformation-cli can be used.

**Comment by wnkz** [comment]
Downgrading does not work and I don't think the issue is what you described. I'll try to open another issue detailing my findings as this is a much broader problem than described here.

**Comment by wesleywh** [comment]
Has anyone come up with a solution to this yet? I'm developing my own custom resource and I have just hit this issue as well. Makes it impossible to run any tests with this or debug my code...

**Comment by jamesoff** [comment]
I was able to work around this by doing the following:

* Clone this repo
* Apply this patch
```diff
diff --git a/src/cloudformation_cli_python_lib/utils.py b/src/cloudformation_cli_python_lib/utils.py
index e4921ff..47f8aab 100644
--- a/src/cloudformation_cli_python_lib/utils.py
+++ b/src/cloudformation_cli_python_lib/utils.py
@@ -119,6 +119,7 @@ class UnmodelledRequest:
     logicalResourceIdentifier: Optional[str] = None
     nextToken: Optional[str] = None
     region: Optional[str] = None
+    awsPartition: Optional[str] = None

     def to_modelled(self, model_cls: Type[BaseModel]) -> BaseResourceHandlerRequest:
         # pylint: disable=protected-access
```
* `cd src; python setup.py sdist`
* Copy the .tar.gz file in `src/dist` to your custom resource project
* Change the line in `requirements.txt` in your custom resource project to `/project/cloudformation-cli-python-lib-2.0.0.tar.gz`

This was enough for me to build a resource which passes `cfn test` and works after being submitted in us-east-1, at least.

**Comment by miparnisari** [comment]
> [aws-cloudformation/cloudformation-cli@b8466be](https://github.com/aws-cloudformation/cloudformation-cli/commit/b8466befe9dffa82f5d6bcb3c071709ceff77163) added new fields to the test event, which are causing an exception when running `cfn test`

Ping @anshikg ?

**Comment by ammokhov** [comment]
closing as being tracked in #112
